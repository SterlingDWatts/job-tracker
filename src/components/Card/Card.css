a.Card {
  text-decoration: none;
  color: black;
  height: fit-content;
}

.Card div {
  background-color: white;
  padding: 5px;
  margin: 5px 0px;
  box-shadow: rgba(9, 30, 66, 0.25) 0px 1px 2px 0px;
}
