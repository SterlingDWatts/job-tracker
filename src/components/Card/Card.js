import React from "react";
import { Link } from "react-router-dom";
import "./Card.css";

export default function Card(props) {
  return (
    <Link className="Card" to={`/application/${props.application.id}`}>
      <div>{props.application.company}</div>
    </Link>
  );
}
