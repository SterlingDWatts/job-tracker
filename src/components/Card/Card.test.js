import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter } from "react-router-dom";
import Card from "./Card";

describe("Card Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Card
          application={{
            id: "1",
            column: [{ id: 0, date: new Date(2020, 4, 13) }],
            company: "Ultimate Software",
            job: "Sotware Engineer 2",
            site: "https://www.ultimatesoftware.com/",
          }}
        />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const wrapper = shallow(
      <Card
        application={{
          id: "1",
          column: [{ id: 0, date: new Date(2020, 4, 13) }],
          company: "Ultimate Software",
          job: "Sotware Engineer 2",
          site: "https://www.ultimatesoftware.com/",
        }}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
