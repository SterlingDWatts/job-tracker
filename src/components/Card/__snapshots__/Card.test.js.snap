// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders the UI as expected 1`] = `
<Link
  className="Card"
  to="/application/1"
>
  <div>
    <div
      className="company"
    >
      Ultimate Software
    </div>
    <div
      className="job"
    >
      Sotware Engineer 2
    </div>
    <div
      className="date"
    >
      May 13, 2020
    </div>
  </div>
</Link>
`;
